runtime/queries/metricsview_aggregation.go

```diff
@@ -579,32 +572,88 @@ func (q *MetricsViewAggregation) buildMetricsAggregationSQL(mv *runtimev1.Metric
 			limitClause,                     // 8
 		)
 	} else {
-		sql = fmt.Sprintf("SELECT %s FROM %s %s %s %s %s %s %s OFFSET %d",
-			strings.Join(selectCols, ", "),
-			safeName(mv.Table),
-			strings.Join(unnestClauses, ""),
-			whereClause,
-			groupClause,
-			havingClause,
-			orderClause,
-			limitClause,
-			q.Offset,
-		)
-	}
+		/*
+			Example:
+			SELECT t.d1, t.d2, t.d3, t2.m1 (SELECT d1, d2, d3, m1 FROM t WHERE ...  GROUP BY d1, d2, d3 HAVING ... ) t LEFT JOIN (
+				SELECT d1, d2, d3, m1 FROM t WHERE ... AND (d4 = 'Safari') GROUP BY d1, d2, d3 HAVING ...
+			)  t2 ON (COALESCE(t.d1, 'val') = COALESCE(t2.d1, 'val') and COALESCE(t.d2, 'val') = COALESCE(t2.d2, 'val') and ...) ORDER BY ...
+
+			This JOIN mirrors functionality of SELECT d1, d2, d3, m1 FILTER (WHERE d4 = 'Safari') FROM t WHERE... GROUP BY d1, d2, d3
+			bacause FILTER cannot be applied for arbitrary measure, ie sum(a)/1000
+		*/
+		if len(q.Measures) == 1 && q.Measures[0].Filter != nil {
```

Comment:
We should return an error if more than 1 measure has a filter (since that's not supported) or if it is used with `pivot_on`